src/Service/NixGenerator.php

```diff
@@ -0,0 +1,392 @@
+                        sha1($cacheUrl),
+                        $package->getDistType()
+                    );
+
+                    // From: `Cache::read()`
+                    $cacheFile = preg_replace('{[^a-z0-9_./]}i', '-', $cacheKey);
+
+                    // Collect package info.
+                    $name = $this->safeNixStoreName($package->getUniqueName());
+
+                    if (false === $sha256 = hash_file('sha256', $this->cacheDir . $cacheFile)) {
+                        $sha256 = $this->fetch($package, $name, $cacheFile);
+                    }
+
+                    yield [
+                        'package' => $package,
+                        'type' => 'cache',
+                        'name' => $name,
+                        'urls' => $urls,
+                        'cacheFile' => $cacheFile,
+                        'sha256' => $sha256,
+                    ];
+
+                    break;
+
+                case 'path':
+                    yield [
+                        'package' => $package,
+                        'type' => 'local',
+                        'name' => $this->safeNixStoreName($package->getName()),
+                        'path' => $package->getDistUrl(),
+                    ];
+
+                    break;
+
+                default:
+                    $this
+                        ->io
+                        ->warning(
+                            sprintf(
+                                "Package '%s' has dist-type '%s' which is not" .
+                                ' supported by the Nixify plugin',
+                                $package->getPrettyName(),
+                                $package->getDistType()
+                            )
+                        );
+
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Generates Nix files based on the lockfile and cache.
+     */
+    public function generate(array $collected): void
+    {
+        $package = $this->composer->getPackage();
+
+        // Build Nix code for cache entries.
+        $cacheEntries = array_filter(
+            $collected,
+            static fn (array $info): bool => 'cache' === $info['type']
+        );
+
+        // Build Nix code for local entries.
+        $localEntries = array_filter(
+            $collected,
+            static fn (array $info): bool => 'local' === $info['type']
+        );
+
+        // If the user bundled Composer, use that in the Nix build as well.
+        $cwd = getcwd() . '/';
+        $composerPath = realpath($_SERVER['PHP_SELF']);
+
+        $composerPath = substr($composerPath, 0, strlen($cwd)) === $cwd
+            ? substr($composerPath, strlen($cwd))
+            : null;
+
+        $jsonData = [
+            'cacheEntries' => $cacheEntries,
+            'composerPath' => $composerPath,
+            'localEntries' => $localEntries,
+            'projectName' => $package->getExtra()['nix-expr-path'] ?? 'composer-project.nix',
```

Comment:
This looks like a mistake. It used to derive from `package->getName()`

Reply:
Fixed.